lib/rubocop/cop/style/empty_string_inside_interpolation.rb

```diff
@@ -74,12 +74,16 @@ def on_interpolation(node)
 
         private
 
-        def empty_if_outcome(node)
-          node.if_branch&.nil_type? || node.if_branch&.value&.empty?
+        def empty_if_outcome?(node)
+          empty_branch_outcome?(node.if_branch)
         end
 
-        def empty_else_outcome(node)
-          node.else_branch&.nil_type? || node.else_branch&.value&.empty?
+        def empty_else_outcome?(node)
+          empty_branch_outcome?(node.else_branch)
+        end
+
+        def empty_branch_outcome?(branch)
+          branch&.nil_type? || (branch&.literal? && branch.value.empty?)
```

Comment:
That's just my two cents.

```suggestion
          return false unless branch

          branch.nil_type? || (branch.literal? && branch.value.empty?)
```

Reply:
Done.